commit ce14c8de3fe3970d24a0928e001a6dc962a79c55 (refs/changes/27/2127/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-07-17T14:54:41-07:00 (2 years, 3 months ago)
    
    OS-6185 use pigz (parallel gunzip uncompression) to speed up imgadm import

diff --git a/src/Makefile b/src/Makefile
index 050daaee..0d712ca0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -382,7 +382,6 @@ install: all $(SUBDIRS)
 	sed -e 's|^#!/usr/bin/env node.*$$|#!/usr/node/bin/node|' \
 	    < node-bunyan/bin/bunyan \
 	    > $(DESTDIR)/usr/bin/bunyan
-	rm -rf $(DESTDIR)/usr/img
 	mkdir -m 0755 -p $(DESTDIR)/usr/img
 	cp img/README.md $(DESTDIR)/usr/img/README.md
 	cp img/package.json $(DESTDIR)/usr/img/package.json
diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index 12f0f6f9..bba44b17 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -6,6 +6,10 @@ Known issues:
   Docker Registry v1, which is now deprecated.
 
 
+## 3.7.3
+
+- OS-6185 use pigz (parallel gunzip uncompression) to speed up imgadm import
+
 ## 3.7.2
 
 - OS-6177 avoid checking content-md5 on imgadm import when a checksum exists on
diff --git a/src/img/package.json b/src/img/package.json
index c9e0ba5b..02fabf3a 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.7.2",
+  "version": "3.7.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/src/img/sbin/chroot-gtar b/src/img/sbin/chroot-gtar
index 15fb1bc1..95959a15 100755
--- a/src/img/sbin/chroot-gtar
+++ b/src/img/sbin/chroot-gtar
@@ -105,8 +105,8 @@ rm -rf $BINDIR
 mkdir $BINDIR
 cp /usr/img/sbin/gtar-unlink-dir $BINDIR/
 CMDS="gtar"
-CMDS="$CMDS cat gzip bzip2 xzcat"
-CMDS="$CMDS bash rm grep sed tee"
+CMDS="$CMDS cat pigz bzip2 xzcat"
+CMDS="$CMDS bash rm grep sed tee nice"
 for cmd in $CMDS; do
     cp $(which $cmd) $BINDIR/
 done
diff --git a/src/img/sbin/gtar-unlink-dir b/src/img/sbin/gtar-unlink-dir
index 64004ebb..5f7d0014 100755
--- a/src/img/sbin/gtar-unlink-dir
+++ b/src/img/sbin/gtar-unlink-dir
@@ -62,21 +62,23 @@ function gtar_unlink_dir
     gtarargs="$4"
 
     local attempts=5
+    local nicecommand="nice -n 5"
     local status=
     local uncompresscat=
-    local uncompressopt=
+    local uncompresscatopts=
+    local uncompressgtaropts=
 
     case $compression in
     none)
         uncompresscat=cat
         ;;
     gzip)
-        uncompresscat=cat
-        uncompressopt=z
+        uncompresscat=pigz
+        uncompresscatopts="-dc -p 2"
         ;;
     bzip2)
         uncompresscat=cat
-        uncompressopt=j
+        uncompressgtaropts=j
         ;;
     xz)
         uncompresscat=xzcat
@@ -93,8 +95,8 @@ function gtar_unlink_dir
         attempts=$(( $attempts - 1 ))
 
         set +o errexit
-        $uncompresscat $tarball \
-            | gtar -C "$subdir" -x${uncompressopt}f - $gtarargs 2>$STDERR_FILE
+        $nicecommand $uncompresscat $uncompresscatopts $tarball \
+            | gtar -C "$subdir" -x${uncompressgtaropts}f - $gtarargs 2>$STDERR_FILE
         status=$?
         set -o errexit
 
